x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/inference_rescore.yml

```diff
@@ -0,0 +1,193 @@
+setup:
+  - skip:
+      features: headers
+  - do:
+      headers:
+        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
+      ml.put_trained_model:
+        model_id: ltr-model
+        body: >
+          {
+            "description": "super complex model for tests",
+            "input": {"field_names": ["cost", "product"]},
+            "inference_config": {
+              "regression": {
+              }
+            },
+            "definition": {
+              "preprocessors" : [{
+                "one_hot_encoding": {
+                  "field": "product",
+                  "hot_map": {
+                    "TV": "type_tv",
+                    "VCR": "type_vcr",
+                    "Laptop": "type_laptop"
+                  }
+                }
+              }],
+              "trained_model": {
+                "ensemble": {
+                  "feature_names": ["cost", "type_tv", "type_vcr", "type_laptop"],
+                  "target_type": "regression",
+                  "trained_models": [
+                    {
+                      "tree": {
+                        "feature_names": [
+                          "cost"
+                        ],
+                        "tree_structure": [
+                          {
+                            "node_index": 0,
+                            "split_feature": 0,
+                            "split_gain": 12,
+                            "threshold": 400,
```

Comment:
Great tests! 
As I understood `threshold` is what separates decision into a left or right nodes. Just for my education, what's `split_feature` and `split_gain` are?

Reply:
`split_feature` is the feature index that we are concerned about on this node. 

`split_gain` does nothing here, but is used to calculate feature importance with SHAP values (if the user desires).